src/main/java/ch/njol/skript/effects/EffHealth.java

```diff
@@ -53,86 +51,85 @@
 
 	static {
 		Skript.registerEffect(EffHealth.class,
-				"damage %livingentities/itemtypes% by %number% [heart[s]][ with fake cause %-damagecause%]",
-				"heal %livingentities% [by %-number% [heart[s]]]",
-				"repair %itemtypes% [by %-number%]");
+			"damage %livingentities/itemtypes% by %number% [heart[s]] [with fake cause %-damagecause%]",
+			"heal %livingentities% [by %-number% [heart[s]]]",
+			"repair %itemtypes% [by %-number%]");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<?> damageables;
 	@Nullable
 	private Expression<Number> damage;
 	private boolean heal = false;
-	@Nullable
-	private Expression<DamageCause> damageCause;
 
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
-		damageables = vars[0];
-		if (ItemStack.class.isAssignableFrom(damageables.getReturnType())) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
+		if (matchedPattern == 0 && exprs[2] != null)
+			Skript.warning("The fake damage cause extension of this effect has no functionality, " +
+				"and will be removed in the future");
+
+		damageables = exprs[0];
+		if (!LivingEntity.class.isAssignableFrom(damageables.getReturnType())) {
 			if (!ChangerUtils.acceptsChange(damageables, ChangeMode.SET, ItemType.class)) {
 				Skript.error(damageables + " cannot be changed, thus it cannot be damaged or repaired.");
 				return false;
 			}
 		}
-		damage = (Expression<Number>) vars[1];
-		heal = (matchedPattern >= 1);
-		
-		if (vars.length >= 3) damageCause = (Expression<DamageCause>) vars[2];
+		damage = (Expression<Number>) exprs[1];
+		heal = matchedPattern >= 1;
+
 		return true;
 	}
 
 	@Override
-	public void execute(final Event e) {
+	public void execute(Event e) {
 		double damage = 0;
 		if (this.damage != null) {
-			final Number n = this.damage.getSingle(e);
-			if (n == null)
+			Number number = this.damage.getSingle(e);
+			if (number == null)
 				return;
-			damage = n.doubleValue();
+			damage = number.doubleValue();
 		}
-		Object[] arr = damageables.getArray(e);
-		if (arr.length > 0 && arr[0] instanceof ItemType) {
-			ItemType[] newarr = new ItemType[arr.length];
-			for (int i = 0; i < arr.length; i++) {
-				ItemStack is = ((ItemType) arr[i]).getRandom();
-				assert is != null;
+		Object[] array = damageables.getArray(e);
+		Object[] newArray = array.clone();
```

Comment:
Since you're manually assinging the values afterwards, cloning the array is kind of pointless
```suggestion
		Object[] newArray = new Object[array.length];
```